abracadabra/src/geometries/sipm.cc

```diff
@@ -19,7 +19,10 @@ G4LogicalVolume* sipm::build() {
   auto act_half_z = act.dz / 2;
   auto vol_body = volume<G4Box>(    name,     mat,     half.x(),     half.y(),     half.z());
   auto vol_act  = volume<G4Box>(act.name, act.mat, act_half_x  , act_half_y  , act_half_z);
-  vol_act->SetSensitiveDetector(new sipm_sensitive("/does/this/matter?", h5_filename));
+
+  sipm_sensitive sens_det{"/does/this/matter?", h5_filename};
+  sens_det.Activate(true);
+  vol_act->SetSensitiveDetector(&sens_det);
```

Comment:
You are handing out a pointer to a locally allocated object which is about to go out of scope (and cease to exist).

Let's take a moment to appreciate that Rust would have refused to compile this code.

I've made the obvious local fix. The tests now pass, rather than segfaulting, so I've pushed it. I still don't know how this fits into the grand scheme of things.